flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetSort.scala

```diff
@@ -71,6 +72,21 @@ class DataSetSort(
     )
   }
 
+  override def estimateRowCount(metadata: RelMetadataQuery): Double = {
+    val inputRowCnt = metadata.getRowCount(this.getInput)
+    if (inputRowCnt == null) {
+      inputRowCnt
+    } else {
+      val rowCount = Math.max(inputRowCnt - limitStart, 0D)
```

Comment:
Returning a cardinality estimate of `0` is not a good idea because all remaining operations might appear to have no costs at all. Rather be conservative and return `1` which is still low but does not invalidate any subsequent costs.